common/djangoapps/student/management/commands/update_anonymous_user_ids.py

```diff
@@ -0,0 +1,110 @@
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import IntegrityError
+from student.models import AnonymousUserId, anonymous_id_for_user
+from submissions.models import StudentItem, ScoreAnnotation
+from openassessment.assessment.models import (
+    AIGradingWorkflow, Assessment, PeerWorkflow, StaffWorkflow, StudentTrainingWorkflow,
+)
+
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.INFO)
+
+
+class Command(BaseCommand):
+    help = """
+    We can now have multiple anonymous user ids for a given user+course.
+    This is useful if we have to change the app's SECRET_KEY, which is used to generate the anonymous user ids.
+    Ref: https://github.com/edx/edx-platform/pull/13717
+
+    However, some models don't reference the Student.AnonymousUserId model directly,
+    and keep their own copies of the anonymous user ID.
+
+    This command runs through an explicit list of those models and fields, and updates
+    them to use the most recently-generated anonymous user ID.
+    """
+
+    def handle(self, *args, **options):
+        """Update the anonymous IDs used by the given models and fields."""
+
+        self.update_anonymous_user_ids()
+        anon_ids_map = self.get_old_to_new_anonymous_user_ids()
+
+        # Update each of the old anonymous IDs with the new one.
+        for (model, field_name) in (
+            (StudentItem, 'student_id'),
+            (ScoreAnnotation, 'creator'),
+            (AIGradingWorkflow, 'student_id'),
+            (Assessment, 'scorer_id'),
+            (PeerWorkflow, 'student_id'),
+            (StaffWorkflow, 'scorer_id'),
+            (StudentTrainingWorkflow, 'student_id'),
+        ):
+            self.update_anonymous_ids(model, field_name, anon_ids_map)
+
+    @staticmethod
+    def update_anonymous_user_ids():
```

Comment:
Minor optional note: `update_anonymous_ids()` and `update_anonymous_user_ids()` are very different methods but have essentially the same name. Calling this one `generate_anonymous_user_ids()` might be better?